lib/reporters/base.js

```diff
@@ -225,11 +226,21 @@ exports.list = function(failures){
  * @api public
  */
 
-function Base(runner) {
+function Base(runner, options, path) {
   var self = this
     , stats = this.stats = { suites: 0, tests: 0, passes: 0, pending: 0, failures: 0 }
     , failures = this.failures = [];
 
+  this.options = options;
+  this.path = path;
+
+  if (path) {
+    if (!fs.createWriteStream) {
```

Comment:
I was thinking about this while working on my branch for plugin support.  

It seems what'd be even more useful is a generic writable stream.  at the risk of ballooning the inheritance tree, I went with keeping the base reporter completely generic.  then, browser-based reporters would use it, and node-based ones would use a subclass `StreamReporter` or something.  they'd just write to their internal stream--whatever that happened to be.  so what if we passed in a writable stream instance instead of a path?  that'd probably eliminate the conditionals below.

you don't have to do this, since my code will conflict with just about everything anyway, but something to think about.